TFG-Labs/custom-shipping-step-by-items
# [WEB-457] Checkout Edit Address Updates

### What problem is this solving?
Currently for the Address Swarm Project - We need to update the following flows
- Add Address Manual
- Add Address Via Search
- Switch to Manual after Adding Address Via Search
- Edit Address

The problem with this approach is: Checkout uses all the same forms for all journeys.

This is reimplementation of the edit address flow in checkout in accordance with:
https://www.notion.so/bashdotcom/Address-Swarm-1-Checkout-Edit-Address-Flow-Updates-88978dc4627d401381449f7037d63f51?pvs=4



### How it works

**EditAddressForm**
-  We have split out 3 components for rendering:
 1. Heading
 2. SaveButton
 3. ContactCard 
 
 Edit Address Form
 -> Takes in an Address 
 -> Only Renders Selected Fields
 
 On Submit: 
 -> Full fields of the the form
 -> Validate the fields (apply UI to them if they are incorrect)
 -> Post The addresses to the UI, Local Storage, orderform, and Masterdata
 -> Switch to the Selected Addresses View

**Delivery Container**
A new section is added with `id="edit-adress-section" ` as a new mounting point for the Edit Form 

**Delivery Controller**
RenderEditAddress: -> takes in an address, and mounts a edit address form into the `edit-adress-section` mounting point

clearEditAddress -> clears the Edit Address from from the mounting point `edit-adress-section`



### How to test it?
https://addressswarmrefactor--thefoschini.myvtex.com/checkout/#/shipping

1. Go to check out from select address view click edit
<img width="547" alt="image" src="https://github.com/user-attachments/assets/51d174b0-5d8f-4384-b8f9-f2cdd08e3511">

2. You are presented with an updated edit address form 
<img width="597" alt="image" src="https://github.com/user-attachments/assets/7c33fdab-238a-4cc4-9a77-c395cbc463c4">

3. On hitting save with invalid data - form validations witll kick in:
<img width="579" alt="image" src="https://github.com/user-attachments/assets/33ac904f-339b-460d-af58-ca1993eeea7c">


4 On Successful submission: the address overview will have selected your updated address and its update name/number will display
<img width="559" alt="image" src="https://github.com/user-attachments/assets/8e516037-ccf7-4d1c-96e7-63dd2c961ba0">

5. The orderforms selected address would have also applied
https://addressswarmrefactor--thefoschini.myvtex.com/api/checkout/pub/orderForm/5280d0421d9f49e89ce85c31821bc537

<img width="567" alt="image" src="https://github.com/user-attachments/assets/3b821a24-ee2e-4623-8a77-9ec5e2b1932f">
<img width="639" alt="image" src="https://github.com/user-attachments/assets/da589304-7267-4226-a8bd-3df10b19718f">






### Related to / Depends on
https://www.notion.so/bashdotcom/Address-Swarm-1-Checkout-Edit-Address-Flow-Updates-88978dc4627d401381449f7037d63f51?pvs=4

#### How does this PR make you feel? :link:
terrible - this sucks to write code in checkout
